tests/Unit/NumericalAlgorithms/Interpolation/Test_PredictedZeroCrossing.cpp

```diff
@@ -10,6 +10,51 @@
 #include "NumericalAlgorithms/Interpolation/LinearLeastSquares.hpp"
 #include "NumericalAlgorithms/Interpolation/PredictedZeroCrossing.hpp"
 
+namespace {
+
+void test_predicted_zero_crossing_datavector() {
+  // Set up random number generator
+  MAKE_GENERATOR(gen);
+  std::uniform_real_distribution<> dist(-10., 10.);
+  std::uniform_real_distribution<> error_dist(-1e-8, 1e-8);
+
+  std::vector<double> expected_zero_crossing_value{};
+  for (size_t i = 0; i < 3; ++i) {
+    expected_zero_crossing_value.push_back(dist(gen));
+  }
+  CAPTURE(expected_zero_crossing_value);
+
+  std::vector<double> slope{};
+  for (size_t i = 0; i < 3; ++i) {
+    slope.push_back(dist(gen));
+  }
+  CAPTURE(slope);
+
+  const DataVector x_values{0., 1., 2.};
+  std::vector<DataVector> y_values{};
+
+  std::vector<double> b{};
+  for (size_t i = 0; i < 3; ++i) {
+    b.push_back(-slope[i] * expected_zero_crossing_value[i]);
+  }
+
+  for (size_t i = 0; i < x_values.size(); ++i) {
+    y_values.push_back(
+        DataVector{slope[i] * x_values[0] + b[i] + error_dist(gen),
+                   slope[i] * x_values[1] + b[i] + error_dist(gen),
+                   slope[i] * x_values[2] + b[i] + error_dist(gen)});
+  }
+
```

Comment:
There are 3 sizes in this test:
  - The size of x_values
  - The size of y_values
  - The size of the DataVector inside of y_values
All 3 of those sizes are the same.  But only 2 of those sizes should be required to be the same.
Please write a test where only 2 of the sizes are the same.

For the use case we care about, x_values will be a set of times (probably in a std::vector<double> or maybe in a std::deque<double>).  Most likely the size of x_values will be something like 10 or 15.

y_values will be a set of DataVectors, where each DataVector is at one of the times in x_values.  So there will be something like 10 or 15 DataVectors.  But each DataVector will have probably 100 to 400 components (the components correspond to the points on a Strahlkorper, and for a Strahlkorper of Ylm resolution L, the number of points is roughly L^2).